### PR TITLE
Don't panic in group tests

### PR DESCRIPTION
**Pull request by cag**
This injects `t *testing.T` into the `GroupTest`s and uses it to allow tests to continue after individual failures, and for tests to report extra info on specific failure cases.

### CODE REVIEW

File: util/test/test.go [hunk 1]
```diff
@@ -50,81 +51,82 @@ func testEmbed(g kyber.Group, rand cipher.Stream, points *[]kyber.Point,
...
 	one := g.Scalar().One()
 	for i := 0; i < N; i++ {
 		s1 := g.Scalar().Pick(rand)
 		s2 := s1.Clone()
 		if !s1.Equal(s2) {
-			panic("Clone didn't create a scalar s2 with same value as s1's.")
+			t.Errorf("Clone didn't work for scalar: %v != %v", s1, s2)
 		}
-		if !s1.Equal(one) {
+		if !s1.Equal(zero) && !s1.Equal(one) {
```
Comment (cag) [inline util/test/test.go]: I forgot to mention that it also considers this edge case correctly.

**Comment by jeffallen** [comment]
Thank you for this.
